doc/syntax.md

```diff
@@ -0,0 +1,476 @@
+
+If there is more than one line specifying a `spike` or `current` port with the same sign, a neuron with multiple receptor types is created. In this case, a `receptor_types` entry is created in the status dictionary, which maps port names to numeric port indices in NEST. The receptor type can then be selected in NEST during [connection setup](http://nest-simulator.org/connection_management/#receptor-types).
+
+### Output
+
+Each neuron model can only send a single type of event. The type of the event has to be given in the `output` block. Currently, however, only spike output is supported.
+```
+output: spike
+```
+
+Please note that this block is *not* terminated with the `end` keyword.
+
+## Synaptic input
+
+NESTML has two dedicated functions to ease the summation of synaptic input.
+
+`curr_sum` is a function that has two arguments. The first is a function *I* of *t* which is either a `shape` function (see [Synaptic response](#synaptic-response)) or a function that is defined by an ODE plus initial values (see [Systems of ODEs](#systems-of-odes)). The second is a `spike` input buffer (see [Synaptic input](#synaptic-input)). `curr_sum` takes every weight in the `spike` buffer and multiplies it with the `shape` function *I*<sub>shape</sub> shifted by it's respective spike time *t<sub>i</sub>*. In mathematical terms, it thus performs the following operation:
+
+<!--- $\large \sum_{t_i\le t, i\in\mathbb{N}}\sum_{w\in\text{spikeweights}} w I_{\text{shape}}(t-t_i)=\sum_{t_i\le t, i\in\mathbb{N}} I_{\text{shape}}(t-t_i)\sum_{w\in\text{spikeweights}} w$ --->
+![equation](https://latex.codecogs.com/svg.latex?%5Clarge%20%5Csum_%7Bt_i%5Cle%20t%2C%20i%5Cin%5Cmathbb%7BN%7D%7D%5Csum_%7Bw%5Cin%5Ctext%7Bspikeweights%7D%7D%20w%20I_%7B%5Ctext%7Bshape%7D%7D%28t-t_i%29%3D%5Csum_%7Bt_i%5Cle%20t%2C%20i%5Cin%5Cmathbb%7BN%7D%7D%20I_%7B%5Ctext%7Bshape%7D%7D%28t-t_i%29%5Csum_%7Bw%5Cin%5Ctext%7Bspikeweights%7D%7D%20w).
+
+When the sum above is used to describe conductances instead of currents, the function `cond_sum` can be used. It does exactly the same as `curr_sum` and can be used in exactly the same way and in the same cases, but makes explicit that the neural dynamics are based on synaptic conductances rather than currents.
+
+For modeling post synaptic responses with delta functions, `curr_sum` and `cond_sum` can be called with the keyword `delta` as first argument instead of a `shape` function.
+
+## Handling of time
+
+To retrieve some fundamental simulation parameters, two special functions are built into NESTML:
+
+* `resolution` returns the current resolution of the simulation in ms. This can be set by the user using the PyNEST function `nest.SetKernelStatus({"resolution": ...})`.
+* `steps` takes one parameter of type `ms` and returns the number of simulation steps in the current simulation resolution.
+
+These functions can be used to implement custom buffer lookup logic, but should be used with care.
+
+## Equations
+
+### Synaptic response
+
```

Comment:
@ingablundell: this section would benefit a lot by having an example and also a figure that demonstrates what such a response looks like.